src/main/java/org/cbio/gdcpipeline/listener/MutationStepListener.java

```diff
@@ -0,0 +1,105 @@
+            isStarted = true;
+        }
+        if (!isStarted) {
+            List<File> maf_files = getMutationFileList();
+            if (!separate_maf.isEmpty()) {
+                if (separate_maf.equalsIgnoreCase("true")) {
+                    //used by metadata step
+                    List<String> maf_filenames = new ArrayList<>();
+                    for(File file : maf_files){
+                        maf_filenames.add(MUTATION_DATA_FILE_PREFIX+file.getName());
+                    }
+                    stepExecution.getJobExecution().getExecutionContext().put("mutation_data_filenames", maf_filenames);
+
+                    //read individually
+                    List<File> mafToProcess = new ArrayList<>();
+                    mafToProcess.add(maf_files.remove(0));
+                    stepExecution.getJobExecution().getExecutionContext().put("maf_files", maf_files);
+                    stepExecution.getExecutionContext().put("mafToProcess", mafToProcess);
+                    stepExecution.getJobExecution().getExecutionContext().put("isStartedMaf", true);
+                } else {
+                    //Read all MAF's together
+                    List<String> mutation_data_filenames = new ArrayList<>();
+                    mutation_data_filenames.add(MUTATION_DEFAULT_MERGED_MAF_FILE);
+                    stepExecution.getJobExecution().getExecutionContext().put("mutation_data_filenames",mutation_data_filenames );
+                    stepExecution.getExecutionContext().put("mafToProcess", maf_files);
+                }
+            }
+        } else {
+            if (!separate_maf.isEmpty()) {
+                if (separate_maf.equalsIgnoreCase("true")) {
+                    List<File> maf_files = (List<File>) stepExecution.getJobExecution().getExecutionContext().get("maf_files");
+                    List<File> mafToProcess = new ArrayList<>();
+                    mafToProcess.add(maf_files.remove(0));
+                    stepExecution.getExecutionContext().put("mafToProcess", mafToProcess);
+                }
+            }
+        }
+    }
+
+    @Override
+    public ExitStatus afterStep(StepExecution stepExecution) {
+        if (!separate_maf.isEmpty()) {
+            if (separate_maf.equalsIgnoreCase("true")) {
+                List<String> mafList = (List<String>) stepExecution.getJobExecution().getExecutionContext().get("maf_files");
+                if (!mafList.isEmpty()) {
+                    return new ExitStatus("CONTINUE");
+                }
+            }
+        }
+        return ExitStatus.COMPLETED;
+    }
+
+    public List<File> getMutationFileList() {
+        List<File> fileList = CommonDataUtil.getFileList(gdcFileMetadatas, CommonDataUtil.GDC_TYPE.MUTATION, sourceDir);
+        List<File> mutationFileList = new ArrayList<>();
+        for (File file : fileList) {
+            File maf =new File(sourceDir,file.getName().replace(".gz", ""));
```

Comment:
I think the data util should return the actual list of files - no other logic should need to be done after calling `getFileList`. Also, if a file is gunzipped at any point, I would update the `Hit` object with the new filename. Maybe you have done that, but just a thought I had at this point in review

Reply:
Actually this should be ignored because the file extraction code in another PR will take care of it..the fucntion is going to return the appropriate file name